test/unit/reductions/moments/test_moments_equalized_odds.py

```diff
@@ -0,0 +1,203 @@
+    assert eqo.pos_basis[0]['-', 'label=False', a1_label] == 0
+    assert eqo.pos_basis[0]['-', 'label=True', a0_label] == 0
+    assert eqo.pos_basis[0]['-', 'label=True', a1_label] == 0
+
+    # Examine the neg_basis_present DF
+    assert len(eqo.neg_basis_present) == 2
+    assert eqo.neg_basis_present[0]
+    assert eqo.neg_basis_present[1]
+
+
+def test_project_lambda_smoke_negatives():
+    eqo = EqualizedOdds()
+
+    events = ['label=False', 'label=True']
+    signs = ['+', '-']
+    labels = ['a', 'b']
+    midx = pd.MultiIndex.from_product(
+        [signs, events, labels],
+        names=[_SIGN, _EVENT, _GROUP_ID])
+
+    df = pd.DataFrame()
+    # Note that the '-' labels are larger
+    df = 0 + pd.Series([1, 2, 11, 19, 1001, 1110, 1230, 1350], index=midx)
+
+    ls = eqo.project_lambda(df)
+
+    expected = pd.DataFrame()
+    expected = 0 + pd.Series([0, 0, 0, 0, 1000, 1108, 1219, 1331], index=midx)
+    assert expected.equals(ls)
+
+
+def test_project_lambda_smoke_positives():
+    # This is a repeat of the _negatives method but with
+    # the '+' indices larger
+    eqo = EqualizedOdds()
+
+    events = ['label=False', 'label=True']
+    signs = ['+', '-']
+    labels = ['a', 'b']
+    midx = pd.MultiIndex.from_product(
+        [signs, events, labels],
+        names=[_SIGN, _EVENT, _GROUP_ID])
+
+    df = pd.DataFrame()
+    # Note that the '-' indices are now smaller
+    df = 0 + pd.Series([200, 300, 100, 600, 4, 5, 6, 7], index=midx)
+
+    ls = eqo.project_lambda(df)
+
+    expected = pd.DataFrame()
+    expected = 0 + pd.Series([196, 295, 94, 593, 0, 0, 0, 0], index=midx)
+    assert expected.equals(ls)
+
+
+def test_signed_weights():
+    eqo = EqualizedOdds()
+    assert eqo.short_name == "EqualizedOdds"
+
+    num_samples_a0 = 10
+    num_samples_a1 = 30
+    num_samples = num_samples_a0 + num_samples_a1
+
+    a0_threshold = 0.2
+    a1_threshold = 0.7
+
+    a0_label = "OneThing"
+    a1_label = "AnotherThing"
+
+    X, Y, A = simple_binary_threshold_data(num_samples_a0, num_samples_a1,
+                                           a0_threshold, a1_threshold,
+                                           a0_label, a1_label)
+
+    # Load up the (rigged) data
+    eqo.load_data(X, Y, sensitive_features=A)
+
+    events = ['label=False', 'label=True']
+    signs = ["+", "-"]
+    labels = [a0_label, a1_label]
+    midx = pd.MultiIndex.from_product(
+        [signs, events, labels],
+        names=[_SIGN, _EVENT, _GROUP_ID])
+
+    lambda_vec = pd.Series([2000, 1000, 4000, 5000, 500, 100, 700, 900], index=midx, name=0)
+
+    lambda_a0_F = 2000 - 500
+    lambda_a0_T = 4000 - 700
+    num_a0_F = int(a0_threshold * num_samples_a0)
+    num_a0_T = num_samples_a0 - num_a0_F
+
+    lambda_a1_F = 1000 - 100
+    lambda_a1_T = 5000 - 900
+    num_a1_F = int(a1_threshold * num_samples_a1)
+    num_a1_T = num_samples_a1 - num_a1_F
+
+    sw_a0_F = (lambda_a0_F + lambda_a1_F) / (1 - sum(Y) / len(Y)) - \
```

Comment:
I've been called out for my glorious multi-line formulas, so I feel the duty to spread the wisdom :-)